config/src/main/java/com/typesafe/config/ConfigDocument.java

```diff
@@ -51,6 +51,24 @@
     ConfigDocument setValue(String path, ConfigValue newValue);
 
     /**
+     * Returns a new ConfigDocument that is a copy of the current ConfigDocument, but with
+     * the value at the desired path removed. If the desired path does not exist in the document,
+     * a copy of the current document will be returned. If there is an array at the root, an exception
+     * will be thrown.
+     *
+     * @param path the path to remove from the document
+     * @return a copy of the ConfigDocument with the desired value removed from the document.
+     */
+    ConfigDocument removeValue(String path);
```

Comment:
docs should say "all values at the desired path" probably (if there are multiple, we remove all presumably)